lightning/src/ln/channelmanager.rs

```diff
@@ -1039,7 +1039,11 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 
 				// OUR PAYMENT!
 				// final_expiry_too_soon
-				if (msg.cltv_expiry as u64) < self.latest_block_height.load(Ordering::Acquire) as u64 + (CLTV_CLAIM_BUFFER + LATENCY_GRACE_PERIOD_BLOCKS) as u64 {
+				// Final node can't rely on checking a CLTV_EXPIRY_DELTA which enforces to be superior to CLTV_CLAIM_BUFFER so make sure
```

Comment:
I don't know what this means. Maybe just say, eg, "We have to have at have some headroom to broadcast on chain if we have the preimage, so make sure we have at least HTLC_FAIL_BACK_BUFFER blocks to go.

Reply:
Took your sentence